apis/v1alpha1/opentelemetrycollector_types.go

```diff
@@ -213,6 +213,9 @@ type OpenTelemetryCollectorSpec struct {
 	// +optional
 	InitContainers []v1.Container `json:"initContainers,omitempty"`
 
+	// AdditionalContainers allows injecting containers to run as sidecars in the pod definition.
+	AdditionalContainers []v1.Container `json:"additionalContainers,omitempty"`
```

Comment:
If we proceed with this PR we should clarify how this is supported e.g. https://github.com/prometheus-operator/prometheus-operator/blob/main/pkg/apis/monitoring/v1/prometheus_types.go#L337

We also should clarify if it is supported for all deployment modes (e.g. deployment, sidecar...)

Reply:
I updated the godoc comment, I added in an explanation, link to additional information about sidecars, and I added a warning about overriding the container that the operator generates. Let me know if this is good enough or if I need to improve it more.